android/KMEA/app/src/main/java/com/tavultesoft/kmea/ModelPickerActivity.java

```diff
@@ -150,349 +200,54 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
   }
 
-  // Each language ID will save to a separate cache
-  private static final String cacheFilename(String languageID) {
-    final String jsonCacheFilename = "jsonLexicalModelsCache";
```

Comment:
Note from here on:  
All async/download/query related functionality in this file was moved to the new `CloudRepository` class to serve as a centralized downloader/updater class.  This code was near-duplicated with that of the `LanguageListActivity` activity's code, and a considerable part of it could be considered near-duplicated with that of the `KeyboardPickerActivity`'s update-check code.